src/appengine/libs/issue_management/oss_fuzz_github.py

```diff
@@ -0,0 +1,196 @@
+# limitations under the License.
+"""Helper functions to file GitHub issues."""
+
+import github
+
+from clusterfuzz._internal.base import utils
+from clusterfuzz._internal.config import db_config
+from clusterfuzz._internal.datastore import data_handler
+from clusterfuzz._internal.metrics import logs
+
+GITHUB_PREFIX = 'https://github.com/'
+
+TESTCASE_REPORT_URL = 'https://{domain}/testcase?key={testcase_id}'
+
+MONORAIL_URL = (
+    'https://bugs.chromium.org/p/oss-fuzz/detail?id={bug_information}')
+
+OSS_FUZZ_ISSUE_URL = 'https://github.com/google/oss-fuzz/issues/new'
+
+ISSUE_TITTLE_TEXT = 'OSS-Fuzz issue {bug_information}'
+
+ISSUE_CONTENT_TEXT = ('OSS-Fuzz has found a bug in this project. Please see '
+                      f'{TESTCASE_REPORT_URL}'
+                      'for details and reproducers.'
+                      '\n\n'
+                      'This issue is mirrored from '
+                      f'{MONORAIL_URL} '
+                      'and will auto-close if the status changes there.'
+                      '\n\n'
+                      'If you have trouble accessing this report, '
+                      f'please file an issue at {OSS_FUZZ_ISSUE_URL}.'
+                      '\n')
+
+ISSUE_ClOSE_COMMENT_TEXT = ('OSS-Fuzz has closed this bug. Please see '
+                            f'{MONORAIL_URL} '
+                            'for details.')
+
+
+def get_issue_title(testcase):
+  """Generate the title of the issue"""
+  return ISSUE_TITTLE_TEXT.format(bug_information=testcase.bug_information)
+
+
+def get_issue_body(testcase):
+  """Generate the body of the issue"""
+  return ISSUE_CONTENT_TEXT.format(
+      domain=data_handler.get_domain(),
+      testcase_id=testcase.key.id(),
+      bug_information=testcase.bug_information)
+
+
+def get_issue_close_comment(testcase):
+  """Generate the closing comment of the issue"""
+  return ISSUE_ClOSE_COMMENT_TEXT.format(
+      bug_information=testcase.bug_information)
+
+
+def _get_access_token():
+  """Get access to GitHub with the oss-fuzz personal access token"""
+  token = db_config.get_value('oss_fuzz_robot_github_personal_access_token')
+  if not token:
+    raise RuntimeError('Unable to access GitHub account and close the issue.')
+  return github.Github(token)
+
+
+def _filing_enabled(testcase):
+  """Check if the project YAML file requires to file a GitHub issue."""
+  require_issue = data_handler.get_value_from_job_definition(
+      testcase.job_type, 'FILE_GITHUB_ISSUE', default='False')
+  return utils.string_is_true(require_issue)
+
+
+def _get_repo(testcase, access):
+  """Get the GitHub repository to file the issue"""
+  repo_url = data_handler.get_value_from_job_definition(testcase.job_type,
+                                                        'MAIN_REPO', '')
+  if not repo_url:
+    logs.log_error('Unable to fetch the MAIN_REPO URL from job definition.')
+    return None
+  repo_name = repo_url.removeprefix(GITHUB_PREFIX)
```

Comment:
Last comment: Not all main_repo values will be github. Can we add a check for this and just exit in such a a case? Log a nomal message here (not error).

Also Line 90 shouldn't be an error log either, as MAIN_REPO is not necessarily always available.